std/testing/mod.ts

```diff
@@ -121,14 +121,26 @@ export function test(
       throw new Error("Missing test function");
     }
     name = t;
+    if (!name) {
+      throw new Error("The name of test case can't be empty");
+    }
+  } else if (typeof t === "function") {
+    fn = t;
+    name = t.name;
+    if (!name) {
+      throw new Error("Test function can't be anonymous");
+    }
   } else {
-    fn = typeof t === "function" ? t : t.fn;
+    fn = t.fn;
+    if (!fn) {
+      throw new Error("Missing test function");
+    }
     name = t.name;
+    if (!name) {
+      throw new Error("The name of test case can't be empty");
+    }
   }
```

Comment:
Do only 2 asserts at the end after `if/else if/else`?

Reply:
Thanks! Added assertion that fn and name aren't empty here.